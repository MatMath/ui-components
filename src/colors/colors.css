:root {

/* -- DASH_GREEN -- */
--dash-green-00: #0E353D;
--dash-green-01: #3E5D64;
--dash-green-02: #6E868B;
--dash-green-03: #9FAEB1;
--dash-green-04: #CFD7D8;
--dash-green-05: #D9DFE0;
--dash-green-06: #F5F7F7;

/* -- DASH_DARKER_GREEN -- */
--dash-darker-green-00: #071A1E;

/* -- MID_GREEN -- */
--mid-green-00: #0E6476;
--mid-green-01: #3E8391;
--mid-green-02: #6EA2AD;
--mid-green-03: #9FC1C8;
--mid-green-04: #CFE0E4;
--mid-green-05: #D9E6E9;

/* -- PINK -- */
--pink-00: #FDB5AF;
--pink-01: #FDC4BF;
--pink-02: #FED3CF;
--pink-03: #FEE1DF;
--pink-04: #FFF0EF;
--pink-05: #FFF3F2;

/* -- ORANGE -- */
--orange-00: #FE5B33;
--orange-01: #FE7C5C;
--orange-02: #FE9D85;
--orange-03: #FFBDAD;
--orange-04: #FFDED6;
--orange-05: #FFE5DE;

/* -- FUNCTIONAL_RED -- */
--functional-red-00: #4E000D;
--functional-red-01: #8B0018;
--functional-red-02: #FF002D;
--functional-red-03: #FFA49F;

/* -- RED -- */
--red-00: #BF0A33;
--red-01: #CC3B5C;
--red-02: #D96C85;
--red-03: #E59DAD;
--red-04: #F2CED6;
--red-05: #F5D8DE;

/* -- GREY -- */
--grey-00: #615B57;
--grey-01: #999592;
--grey-02: #CCC7C2;
--grey-03: #D6D2CE;
--grey-04: #E0DDDA;
--grey-05: #EBE9E7;
--grey-06: #F5F4F3;

/* -- BLACK -- */
--black: #000000;

/* -- WHITE -- */
--white: #FFFFFF;

/* -- VALIDATOR_GREEN -- */
--validator-green: #20B422;

/* -- TRANSPARENT -- */
--transparent: transparent;
}
