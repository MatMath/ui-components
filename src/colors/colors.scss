/* -- DASH_GREEN -- */
$dash-green-00: #0e353d;
$dash-green-01: #3e5d64;
$dash-green-02: #6e868b;
$dash-green-03: #9faeb1;
$dash-green-04: #cfd7d8;
$dash-green-05: #d9dfe0;
$dash-green-06: #f5f7f7;

/* -- DASH_DARKER_GREEN -- */
$dash-darker-green-00: #071a1e;

/* -- MID_GREEN -- */
$mid-green-00: #0e6476;
$mid-green-01: #3e8391;
$mid-green-02: #6ea2ad;
$mid-green-03: #9fc1c8;
$mid-green-04: #cfe0e4;
$mid-green-05: #d9e6e9;

/* -- PINK -- */
$pink-00: #fdb5af;
$pink-01: #fdc4bf;
$pink-02: #fed3cf;
$pink-03: #fee1df;
$pink-04: #fff0ef;
$pink-05: #fff3f2;

/* -- ORANGE -- */
$orange-00: #fe5b33;
$orange-01: #fe7c5c;
$orange-02: #fe9d85;
$orange-03: #ffbdad;
$orange-04: #ffded6;
$orange-05: #ffe5de;

/* -- FUNCTIONAL_RED -- */
$functional-red-00: #4e000d;
$functional-red-01: #8b0018;
$functional-red-02: #ff002d;
$functional-red-03: #ffa49f;

/* -- RED -- */
$red-00: #bf0a33;
$red-01: #cc3b5c;
$red-02: #d96c85;
$red-03: #e59dad;
$red-04: #f2ced6;
$red-05: #f5d8de;

/* -- GREY -- */
$grey-00: #615b57;
$grey-01: #999592;
$grey-02: #ccc7c2;
$grey-03: #d6d2ce;
$grey-04: #e0ddda;
$grey-05: #ebe9e7;
$grey-06: #f5f4f3;

/* -- BLACK -- */
$black: #000;

/* -- WHITE -- */
$white: #fff;

/* -- VALIDATOR_GREEN -- */
$validator-green: #20b422;

/* -- ACCESSIBLE_VALIDATOR_GREEN -- */
$accessible-validator-green: #278637;

/* -- ACCESSIBLE_VALIDATOR_RED -- */
$accessible-validator-red: #e00635;

/* -- TRANSPARENT -- */
$transparent: transparent;
