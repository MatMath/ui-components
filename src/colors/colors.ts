/* -- DASH_GREEN -- */
export const dashGreen00 = '#0E353D';
export const dashGreen01 = '#3E5D64';
export const dashGreen02 = '#6E868B';
export const dashGreen03 = '#9FAEB1';
export const dashGreen04 = '#CFD7D8';
export const dashGreen05 = '#D9DFE0';
export const dashGreen06 = '#F5F7F7';

/* -- DASH_DARKER_GREEN -- */
export const dashDarkerGreen00 = '#071A1E';

/* -- MID_GREEN -- */
export const midGreen00 = '#0E6476';
export const midGreen01 = '#3E8391';
export const midGreen02 = '#6EA2AD';
export const midGreen03 = '#9FC1C8';
export const midGreen04 = '#CFE0E4';
export const midGreen05 = '#D9E6E9';

/* -- PINK -- */
export const pink00 = '#FDB5AF';
export const pink01 = '#FDC4BF';
export const pink02 = '#FED3CF';
export const pink03 = '#FEE1DF';
export const pink04 = '#FFF0EF';
export const pink05 = '#FFF3F2';

/* -- ORANGE -- */
export const orange00 = '#FE5B33';
export const orange01 = '#FE7C5C';
export const orange02 = '#FE9D85';
export const orange03 = '#FFBDAD';
export const orange04 = '#FFDED6';
export const orange05 = '#FFE5DE';

/* -- FUNCTIONAL_RED -- */
export const functionalRed00 = '#4E000D';
export const functionalRed01 = '#8B0018';
export const functionalRed02 = '#FF002D';
export const functionalRed03 = '#FFA49F';

/* -- RED -- */
export const red00 = '#BF0A33';
export const red01 = '#CC3B5C';
export const red02 = '#D96C85';
export const red03 = '#E59DAD';
export const red04 = '#F2CED6';
export const red05 = '#F5D8DE';

/* -- GREY -- */
export const grey00 = '#615B57';
export const grey01 = '#999592';
export const grey02 = '#CCC7C2';
export const grey03 = '#D6D2CE';
export const grey04 = '#E0DDDA';
export const grey05 = '#EBE9E7';
export const grey06 = '#F5F4F3';

/* -- BLACK -- */
export const black = '#000000';

/* -- WHITE -- */
export const white = '#FFFFFF';

/* -- VALIDATOR_GREEN -- */
export const validatorGreen = '#20B422';

/* -- ACCESSIBLE_VALIDATOR_GREEN -- */
export const accessibleValidatorGreen = '#278637';

/* -- ACCESSIBLE_VALIDATOR_RED -- */
export const accessibleValidatorRed = '#e00635';

/* -- TRANSPARENT -- */
export const transparent = 'transparent';
