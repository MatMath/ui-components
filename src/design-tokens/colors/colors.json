{
  "dash-green": {
    "00": "#0E353D",
    "01": "#3E5D64",
    "02": "#6E868B",
    "03": "#9FAEB1",
    "04": "#CFD7D8",
    "05": "#D9DFE0",
    "06": "#F5F7F7"
  },
  "dash-darker-green": {
    "00": "#071A1E"
  },
  "mid-green": {
    "00": "#0E6476",
    "01": "#3E8391",
    "02": "#6EA2AD",
    "03": "#9FC1C8",
    "04": "#CFE0E4",
    "05": "#D9E6E9"
  },
  "pink": {
    "00": "#FDB5AF",
    "01": "#FDC4BF",
    "02": "#FED3CF",
    "03": "#FEE1DF",
    "04": "#FFF0EF",
    "05": "#FFF3F2"
  },
  "orange": {
    "00": "#FE5B33",
    "01": "#FE7C5C",
    "02": "#FE9D85",
    "03": "#FFBDAD",
    "04": "#FFDED6",
    "05": "#FFE5DE"
  },
  "functional-red": {
    "00": "#4E000D",
    "01": "#8B0018",
    "02": "#FF002D",
    "03": "#FFA49F"
  },
  "red": {
    "00": "#BF0A33",
    "01": "#CC3B5C",
    "02": "#D96C85",
    "03": "#E59DAD",
    "04": "#F2CED6",
    "05": "#F5D8DE"
  },
  "warning": {
    "00": "#BF2600"
  },
  "grey": {
    "00": "#615B57",
    "01": "#999592",
    "02": "#CCC7C2",
    "03": "#D6D2CE",
    "04": "#E0DDDA",
    "05": "#EBE9E7",
    "06": "#F5F4F3"
  },
  "black": "#000000",
  "white": "#FFFFFF",
  "validator-green": "#20B422",
  "accessible-validator-green": "#278637",
  "accessible-validator-red": "#e00635",
  "transparent": "transparent"
}
