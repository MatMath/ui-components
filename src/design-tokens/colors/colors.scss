/* -- PRIMARY -- */
$primary-100: #cce5ea;
$primary-300: #8cc4d0;
$primary-500: #007c97;
$primary-600: #005d71;
$primary-700: #003945;
$primary-800: #002e38;
$primary-900: #062931;

/* -- SECONDARY -- */
$secondary-300: #ffc1a3;
$secondary-500: #ff5412;
$secondary-700: #b34015;
$secondary-900: #823214;

/* -- SUCCESS -- */
$success-300: #c1eac1;
$success-500: #20b422;
$success-700: #157917;
$success-900: #105c11;

/* -- DANGER -- */
$danger-300: #ffe3dd;
$danger-500: #ff002d;
$danger-700: #8b0018;
$danger-900: #4e000d;

/* -- WARNING -- */
$warning-300: #fffac1;
$warning-500: #ffe900;
$warning-700: #595200;
$warning-900: #2e2a00;

/* -- INFO -- */
$info-300: #bdedff;
$info-500: #00b8ff;
$info-700: #007dad;
$info-900: #00e042;

/* -- LIGHT -- */
$light-100: #ffffff;
$light-200: #fafafa;
$light-300: #f6f6f6;
$light-400: #e7e7e7;
$light-500: #d6d6d6;
$light-600: #9b9b9b;
$light-700: #474747;
$light-900: #191919;
