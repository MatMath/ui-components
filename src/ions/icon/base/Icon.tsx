import styled from 'styled-components';
import { Props as IconProps } from '@ions/icon/types';
import { dashGreen00 } from '@colors';

export type ColorProps = 'color' | 'hoverColor' | 'activeColor';
interface Props extends Omit<IconProps, 'title' | 'children'> {
  animated?: boolean;
}

const getWidth = ({ width, size }: Props) => `${size || width || 20}px`;
const getHeight = ({ height, size }: Props) => `${size || height || 20}px`;
const getColor = (propName: ColorProps = 'color') => (props: Props): string =>
  props.animated ? 'none' : props[propName] || props['color'] || dashGreen00;

export const Icon = styled.svg.attrs<Props>(({ viewBox }: Props) => ({
  viewBox: viewBox || '0 0 20 20',
  version: '1.1',
  xmlns: 'http://www.w3.org/2000/svg'
}))`
  display: flex;
  width: ${getWidth};
  height: ${getHeight};
  cursor: ${({ disabled }): string => (disabled ? 'not-allowed' : 'inherit')};
  outline: none;
  border-radius: 4px;
  fill: ${getColor()};
  opacity: ${({ disabled }): number => (disabled ? 0.5 : 1)};
  transition: ${({ animated }): string =>
    animated ? 'none' : 'all 0.1s ease-in'};
  &:hover {
    fill: ${getColor('hoverColor')};
  }
  &:active {
    fill: ${getColor('activeColor')};
  }
`;
