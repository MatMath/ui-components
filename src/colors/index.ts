import { BasicColor, ColorExclude } from '@typings/Color.d';
import * as colors from '@design-tokens/colors.json';

export const $primary: ColorExclude<'200' | '400'> = colors.primary;

export const $secondary: BasicColor = colors.secondary;

export const $success: BasicColor = colors.success;

export const $danger: BasicColor = colors.danger;

export const $warning: BasicColor = colors.warning;

export const $info: BasicColor = colors.info;

export const $light: ColorExclude<'800'> = colors.light;

export const $transparent = 'transparent';
