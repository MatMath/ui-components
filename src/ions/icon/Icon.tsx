import styled from 'styled-components';
import { Props as IconProps } from '@atoms/svg/base/SvgIcon';

export type ColorProps = 'color' | 'hoverColor' | 'activeColor';
interface Props extends Omit<IconProps, 'title' | 'children'> {
  animated?: boolean;
}

const getSize = ({ size = 20 }) => `${size}px`;
const getColor = (propName: ColorProps = 'color') => (props: Props): string =>
  props.animated ? 'none' : props[propName] || props['color'] || 'inherit';

const Icon = styled.svg.attrs<Props>(({ viewBox }: Props) => ({
  viewBox: viewBox || '0 0 20 20',
  version: '1.1',
  xmlns: 'http://www.w3.org/2000/svg'
}))`
  display: flex;
  width: ${getSize};
  height: ${getSize};
  cursor: ${({ disabled }): string => (disabled ? 'not-allowed' : 'pointer')};
  outline: none;
  border-radius: 4px;
  fill: ${getColor()};
  opacity: ${({ disabled }): number => (disabled ? 0.5 : 1)};
  transition: ${({ animated }): string =>
    animated ? 'none' : 'all 0.1s ease-in'};
  &:hover {
    fill: ${getColor('hoverColor')};
  }
  &:active {
    fill: ${getColor('activeColor')};
  }
`;

export default Icon;
