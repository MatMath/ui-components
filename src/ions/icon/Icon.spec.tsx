import * as React from 'react';
import { mount } from 'enzyme';
import 'jest-styled-components';
import Icon from './Icon';
import { $light } from '@globals/colors';

describe('<Icon>', () => {
  describe('global render', () => {
    it('should render', () => {
      const IconWrapper = mount(<Icon />);

      expect(IconWrapper).toMatchSnapshot();
    });

    describe('component style', () => {
      it('should set a default style if none is provided', () => {
        const IconWrapper = mount(<Icon />);

        expect(IconWrapper).toHaveStyleRule('fill', $light[700]);
        expect(IconWrapper).toHaveStyleRule('cursor', 'pointer');
        expect(IconWrapper).toHaveStyleRule('width', '2em');
        expect(IconWrapper).toHaveStyleRule('height', '2em');
      });

      it('should set the provided style', () => {
        const IconWrapper = mount(<Icon size={4} color='red' />);

        expect(IconWrapper).toHaveStyleRule('fill', 'red');
        expect(IconWrapper).toHaveStyleRule('width', '4em');
        expect(IconWrapper).toHaveStyleRule('height', '4em');
      });

      it('should handle disabled mode', () => {
        const IconWrapper = mount(<Icon disabled />);

        expect(IconWrapper).toHaveStyleRule('opacity', '0.2');
        expect(IconWrapper).toHaveStyleRule('cursor', 'not-allowed');
      });
    });
  });
});
