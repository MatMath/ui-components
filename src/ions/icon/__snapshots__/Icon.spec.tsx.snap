// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> global render should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 1.5em;
  height: 1.5em;
  cursor: pointer;
  outline: none;
  border-radius: 4px;
  fill: inherit;
  opacity: 1;
  -webkit-transition: all 0.1s ease-in;
  transition: all 0.1s ease-in;
}

.c0:hover {
  fill: inherit;
}

.c0:active {
  fill: inherit;
}

<styled.svg>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [
          [Function],
        ],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  display: flex;
  width: ",
            [Function],
            ";
  height: ",
            [Function],
            ";
  cursor: ",
            [Function],
            ";
  outline: none;
  border-radius: 4px;
  fill: ",
            [Function],
            ";
  opacity: ",
            [Function],
            ";
  transition: ",
            [Function],
            ";
  &:hover {
    fill: ",
            [Function],
            ";
  }
  &:active {
    fill: ",
            [Function],
            ";
  }
",
          ],
        },
        "displayName": "styled.svg",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "svg",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <svg
      className="c0"
      version="1.1"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    />
  </StyledComponent>
</styled.svg>
`;
